### PR TITLE
Custom Animation Styles for WoltModalSheet by using 

### PR DESCRIPTION
**Pull request by ulusoyca**
## Description

This PR allows developers to customize the animation styles for both scrolling and pagination within the `WoltModalSheet` by using the new `WoltModalSheetAnimationStyle` class.

The core of this feature is the introduction of the `WoltModalSheetAnimationStyle` class, which encompasses two key aspects:

- **Pagination Animations**: Customize how the modal sheet animates during page transitions. This includes settings for main content opacity, modal sheet height transition, and the sticky action bar (SAB) opacity, among others.
- **Scrolling Animations**: Defines animations related to scrolling within the modal sheet. If the the top visibility is set to change as the user scrolls, the `WoltModalSheetScrollAnimationStyle` covers hero image scaling, top bar translation, and more.

These animations are applied through the `WoltModalSheetThemeData`.
### Example Configuration

```dart
WoltModalSheetThemeData(
  animationStyle: WoltModalSheetAnimationStyle(
    paginationAnimationStyle: WoltModalSheetPaginationAnimationStyle(
      mainContentIncomingOpacityCurve: const Interval(
        150 / 350,
        350 / 350,
        curve: Curves.linear,
      ),
      modalSheetHeightTransitionCurve: const Interval(
        0 / 350,
        300 / 350,
        curve: Curves.fastOutSlowIn,
      ),
      incomingSabOpacityCurve: const Interval(
        100 / 350,
        300 / 350,
        curve: Curves.linear,
      ),
      // Define additional pagination animation styles as needed.
    ),
    scrollAnimationStyle: WoltModalSheetScrollAnimationStyle(
      heroImageScaleStart: 1.0,
      heroImageScaleEnd: 0.9,
      topBarTitleTranslationYInPixels: 8.0,
      topBarTranslationYInPixels: 4.0,
      // Define additional scroll animation styles as needed.
    ),
  ),
),
```

| Pagination | Scrolling |
|--------|--------|
| <video src="https://github.com/woltapp/wolt_modal_sheet/assets/13782003/8f00650b-83d8-41d0-aa72-13c368f03036"> |  <video src="https://github.com/woltapp/wolt_modal_sheet/assets/13782003/e848a166-a043-45f8-83b8-ccbfdb7cd554"> | 

## Related Issues

* https://github.com/woltapp/wolt_modal_sheet/issues/131
* https://github.com/woltapp/wolt_modal_sheet/issues/125 (This is fixed by disabling the image scaling by default.)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors.
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] The package compiles with the minimum Flutter version stated in the [pubspec.yaml](https://github.com/woltapp/wolt_modal_sheet/blob/main/pubspec.yaml#L8)

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[Contributor Guide]: https://github.com/woltapp/wolt_modal_sheet/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart [hunk 1]
```diff
@@ -68,6 +74,8 @@ class _HeroImageFlowDelegate extends FlowDelegate {
   /// The height of the hero image.
   final double heroImageHeight;
 
+  final WoltModalSheetScrollAnimationStyle scrollAnimationStyle;
```
Comment (TahaTesser) [inline lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart:77]: Missing doc

File: lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart [hunk 2]
```diff
@@ -31,13 +33,16 @@ class WoltModalSheetHeroImage extends StatelessWidget {
...
```
Comment (TahaTesser) [inline lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart:36]: Missing doc

**Review by TahaTesser — APPROVED** [review]
LGTM with some nits!

File: lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart (hunk 1 above)
Comment (ulusoyca) [reply lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart]: Done 5def56efa78768e9333eaa867cb06c9a593fb33c

File: lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart (hunk 2 above)
Comment (ulusoyca) [reply lib/src/content/components/main_content/wolt_modal_sheet_hero_image.dart]: Done 5def56efa78768e9333eaa867cb06c9a593fb33c

**Pull request merged**